odoc.opam

```diff
@@ -47,6 +47,7 @@ depends: [
   "conf-jq" {with-test}
 
   "bisect_ppx" {with-test & = "2.5.0"}
+  "ppx_expect" {with-test}
```

Comment:
On line 46, `sexplib0` won't be installed on ocaml version below `4.04.1`. Either this need to change or, if it really can't work below that version, we need a `enabled_if` condition on the parser tests.

Reply:
that's what (optional) above is for.

Reply:
The problem with `(optional)` is that we won't know if the tests are run. `enabled_if` seems more robust. I'm also suggesting to remove this constraint and see.

Reply:
I'm not sure I understand why it's more robust - wouldn't we have the same problem of not knowing if the tests are run with both constructs? Is it just that it's more explicitly about the ocaml version?

Reply:
If some versions can't run some tests, that's fine but that should be done with `enabled_if` to ensure it's correct and that we really want that (eg. one pure condition instead of implicit assumption about the environment).
`(optional)` dependencies will implicitly disable some tests depending on the current state of my dev environment (may be out of sync from the changing opam description). I almost wrote that tests don't run in my review.
Also, I may have `sexplib0` installed for an other reason, triggering tests that will fails (`enabled_if` is needed).

Reply:
I find https://github.com/ocaml/odoc/pull/642#discussion_r595298675 sensible, albeit, ` (enabled_if false)` still won’t warn if `ppx_expect` is not installed, and ` (enabled_if true)`, warns that `sexplib0` requires `>=4.04.1`

Reply:
> Is it just that it's more explicitly about the ocaml version?

I think so

Reply:
Removing the `(optional)` is a good thing but now we have broken tests. You need to add `(enabled_if (>= %{ocaml_version} 4.04.1))`